### PR TITLE
[aws] Combine Extensions.AWS, Instrumentation.AWS, & Instrumentation.AWSLambda into a single tag

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

* Use the same tag prefix (`MinVerTagPrefix`) for `OpenTelemetry.Extensions.AWS`, `OpenTelemetry.Instrumentation.AWS`, & `OpenTelemetry.Instrumentation.AWSLambda`.

* Add STJ reference at `6.0.10` / `8.0.5` for `OpenTelemetry.Instrumentation.AWSLambda` since it uses the source generator.

## Details

@srprash @ppittle @Oberon00 @rypdal

My goal here is to resolve the last build breaks due to vulnerable STJ package versions:

```
OpenTelemetry.Instrumentation.AWSLambda.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 6.0.0 has a known high severity vulnerability
OpenTelemetry.Instrumentation.AWS.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 6.0.0 has a known high severity vulnerability
```

The reason this is happening is `OpenTelemetry.Instrumentation.AWS` and `OpenTelemetry.Instrumentation.AWSLambda` reference `OpenTelemetry.Extensions.AWS` through NuGet, they don't use `ProjectReference`.

There isn't anything necessarily wrong with that, but it is a code smell IMO. Tightly coupled projects should be released together.

Examples:

* [OpenTelemetry.Instrumentation.AspNet](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.AspNet/OpenTelemetry.Instrumentation.AspNet.csproj#L9) & [OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/f6bd1a7d391481ac9359e5c30177c69490ce1ee6/src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule.csproj#L8)

* [OpenTelemetry.PersistentStorage.Abstractions](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/f6bd1a7d391481ac9359e5c30177c69490ce1ee6/src/OpenTelemetry.PersistentStorage.Abstractions/OpenTelemetry.PersistentStorage.Abstractions.csproj#L9) & [OpenTelemetry.PersistentStorage.FileSystem](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/f6bd1a7d391481ac9359e5c30177c69490ce1ee6/src/OpenTelemetry.PersistentStorage.FileSystem/OpenTelemetry.PersistentStorage.FileSystem.csproj#L9)

What this PR is doing is using the same `MinVerTagPrefix` for `OpenTelemetry.Extensions.AWS`, `OpenTelemetry.Instrumentation.AWS`, & `OpenTelemetry.Instrumentation.AWSLambda` so they will always be versioned and released together.

We could just put out a NuGet for `OpenTelemetry.Extensions.AWS` to resolve the issue but I feel this is better design and overall better for users. Less chance for mismatched versions and conflicts for the coupled projects and their dependencies.

Please approve this PR if this works for AWS group!

PS: We could version _all_ of the AWS packages together. Might simplify the versioning/upgrade story for everything. But I only did the coupled ones here. Happy to do them all if that is something AWS group wants.

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)

### CODE REVIEW

**Review by Kielek — APPROVED** [review]
LGTM. I would wait for some feedback from component owners before merge.

**Comment by ppittle** [comment]
Given our recent discussion in the SIG about release strategy, I agree this change make sense.

I'd still like to review the other AWS packages - it may yet make sense for those to release separately.

Thanks for the change and write up!

**Review by ppittle — APPROVED** [review]

**Pull request merged**
